rerun_py/rerun_sdk/rerun/datatypes/vec3d_ext.py

```diff
@@ -24,5 +26,7 @@ def native_to_pa_array_override(data: Vec3DArrayLike, data_type: pa.DataType) ->
             if isinstance(data, Sequence):
                 data = [np.array(p.xyz) if isinstance(p, Vec3D) else p for p in data]
 
-        points = np.asarray(data, dtype=np.float32).reshape((-1,))
+        points = np.asarray(data, dtype=np.float32)
+        validate_vector_shape(points.shape, 3)
+        points = points.reshape((-1,))
```

Comment:
[Parse, don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)

```suggestion
        points = flat_np_array_from_vector_shape(data, 3, type=np.float32)
```

here and everywhere - saves some code, and makes the intention clearer imho